### DISCUSSION TITLE
XRMToolbox - Tool - Sync Filter Manager - How to Select Subset of Users using specific User View and bulk enable-disable user filter

### DISCUSSION THREAD

**Discussion by nickgsoft** [MscrmTools/XrmToolBox#1131] [Ideas]
XRMToolbox - Tool - Sync Filter Manager - 

How to Select Subset of Users using specific User View and bulk enable-disable user filter?

Is there a possibility to add such a feature or does that exist already?

**Comment by MscrmTools** [comment]
Can you explain a bit more your scenario?

**Reply by nickgsoft** [reply]
@MscrmTools 
This Plugin allows only two option 
"For Specific Users"  which allow only one user at a time search
and 
"For All Users".  this pulls all users which takes long time to load 

What it is missing is within "For Specific Users" screen, if there is a way to allow User View (Advanced Find - FetchXML) Selection to bring few subset of users, then it will reduce time to load and make easier Bulk Enable-Disable operations. Thoughts?

**Reply by MscrmTools** [reply]
Understood!
Will try to work on it soon

**Reply by nickgsoft** [reply]
@MscrmTools 
So are you saying this is valid feature and do not exist at the moment and  can be developed, correct?

**Reply by MscrmTools** [reply]
Correct

**Reply by nickgsoft** [reply]
@MscrmTools  

- One more quick question, If this Plugin can pull all users personalization Filters, is there any Dataverse SQL query can be written to pull How many users have specific Appointment view filter is active? 

The reason i am asking is this tool do not have a way to export such list either?

**Reply by MscrmTools** [reply]
There is not such feature, no.
I would need to review my code to remember how the filter is applied to users and so how to export the information.
I will let you know next week

**Reply by nickgsoft** [reply]
@MscrmTools  
FYI : Donation effort is underway to XRMTOOLBOX paypal account

**Reply by MscrmTools** [reply]
Thank you very much 😊

**Reply by nickgsoft** [reply]
@MscrmTools  

- Let me know if i can help in any ways coding-analysis-testing-BA-smoke testing and such.....

**Reply by MscrmTools** [reply]
I will surely ask for some testing once I will have implemented the user selection and/or info export 
I should have something to test during next week

**Comment by MscrmTools** [comment]
@nickgsoft, can you test this version, please?
[MscrmTools.SyncFilterManager.zip](https://github.com/MscrmTools/XrmToolBox/files/8326143/MscrmTools.SyncFilterManager.zip)

This is for an improvement to the users selection

**Comment by nickgsoft** [comment]
@MscrmTools 

![image](https://user-images.githubusercontent.com/12518671/159708107-54334d75-98b8-4e8e-81af-f5dfdbcf1b7b.png)

- While reviewing, i do see views screen shows up and able to retrieve system views only, I did not had personal views, then i created personal view on the test environment and published all as well, after that i connected the XRMToolbox to that environment and still i could not see my personal views on Sync Filter Manager "Select users screen". I attached the screenshot. 
- 
![image](https://user-images.githubusercontent.com/12518671/159708060-788f1840-8658-492a-9fd7-277c4b35b2ab.png)


There are couple of errors i encountered:
1) Below is one error i was able to get why exiting the userview screen. 


`See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at MscrmTools.SyncFilterManager.Controls.CrmUserList.bw_RunWorkerCompleted(Object sender, RunWorkerCompletedEventArgs e)
   at System.ComponentModel.BackgroundWorker.OnRunWorkerCompleted(RunWorkerCompletedEventArgs e)

2) Another Error "Error while searching users: attempted divide by zero" Attached is the scrrenshot.

![image](https://user-images.githubusercontent.com/12518671/159708088-58ed2cf3-4ccf-4d58-bab9-abf9acf9b32c.png)

**Reply by MscrmTools** [reply]
Thank you for the feedback. 
From what you see (sorry for the bugs), is this what you expected as UI for having a more precise user selection?

**Reply by nickgsoft** [reply]
yes the UI is perfect.,.... Thank you for quickly taking action..

**Comment by MscrmTools** [comment]
ok, new version is coming soon in Tools Library

**Reply by nickgsoft** [reply]
@MscrmTools  Looking forward to it...can't wait... 👍

**Reply by MscrmTools** [reply]
Actually, it should be available

**Reply by nickgsoft** [reply]
@MscrmTools 
![image](https://user-images.githubusercontent.com/12518671/160714112-41e3fa02-3c0a-412c-9935-d3621729a9f7.png)

I see it

**Reply by nickgsoft** [reply]
@MscrmTools  
for what ever reason, i still get same errors and could not see personal view on filter.

**Reply by MscrmTools** [reply]
Really strange... everything works well on my side...
Are you system admin?

**Reply by nickgsoft** [reply]
@MscrmTools  i am admin and i also removed all installed plugins from roaming/xrmtoolbox directory and clean and redid that again few times, but no luck. 

for what ever reason, i still could not see list of personal views on filter.

![image](https://user-images.githubusercontent.com/12518671/161319553-904eec54-a681-4b3d-82aa-b865fd5970dd.png)

**Reply by nickgsoft** [reply]
@MscrmTools  
i revisited and tested the tool by removing all the versions, DLL, references, previous copies and cache from my PC, and then tried and then i can see it works. 

![image](https://user-images.githubusercontent.com/12518671/162087113-912e992c-0e0d-470b-b260-ef0e031ed0b3.png)

Donation is on the way! Thanks a lot for this quick effort and helping the community.

**Comment by MscrmTools** [comment]
@nickgsoft wow! this is very generous! Thank you very much! If you want XrmToolBox stickers or buttons, just tell me and I'll send some
